test/config/integration/server_ssl.json

```diff
@@ -99,7 +99,7 @@
       "connect_timeout_ms": 5000,
       "ssl_context": {
         "ca_cert_file": "{{ test_rundir }}/test/config/integration/certs/upstreamcacert.pem",
-        "verify_subject_alt_name": [ "foo.lyft.com" ]
+        "verify_subject_alt_name": [ "*.lyft.com" ]
```

Comment:
I'd prefer if you add new test cases, so that we have both exact match and wildcard covered.